### PR TITLE
Enable Airflow to run a dag using an existing cluster

### PR DESCRIPTION
**Pull request by Harshith-umesh**
### Description
Adds an option to pass OC login creds to allow the execution of an airflow dag.

This PR adds a new platform called "prebuilt" which has 2 variants, control-plane and data-plane. These 2 dags can only be triggered manually and need to be triggered using the "Trigger with config" option where you need to fill the following json file :
```
{
    "KUBEUSER": "<Enter openshift cluster-admin username>",
    "KUBEPASSWORD": "<Enter openshift cluster password>",
    "KUBEURL": "<Enter cluster URL>"
}
```
This PR also fixes the failing must-gather task.
Link to a successful dag run using a prebuilt cluster : http://harshith-umesh-prebuilt-airflow.apps.sailplane.perf.lab.eng.rdu2.redhat.com/graph?dag_id=4.x-prebuilt-data-plane(same password as main)

### CODE REVIEW

File: dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            **self.aws_creds,
+            **self.gcp_creds,
+            **self.azure_creds,
+            **self.openstack_creds,
+            **self.rosa_creds,
+            **self.rhacs_creds,
+            **{ "es_server": var_loader.get_secret('elasticsearch'),
+                "thanos_receiver_url": var_loader.get_secret('thanos_receiver_url'),
+                "loki_receiver_url": var_loader.get_secret('loki_receiver_url') }
+        }
```
Comment (whitleykeith) [inline dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py:56]: do we need all of this? I don't think prebuilt needs any of these secrets

File: dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        self.env = {
+            "SSHKEY_TOKEN": self.config['sshkey_token'],
+            "OPENSHIFT_CLUSTER_NAME": self.cluster_name,
+            "KUBECONFIG_NAME": f"{self.release_name}-kubeconfig",
+            "KUBEADMIN_NAME": f"{self.release_name}-kubeadmin",
+            "OPENSHIFT_INSTALL_PULL_SECRET": self.ocp_pull_secret,
+            "GCP_MANAGED_SERVICES_TOKEN": self.rogcp_creds,
+            "AWS_REGION": self.config['aws_region_for_openshift'],
```
Comment (whitleykeith) [inline dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py:65]: same with some of these. Shouldn't we only need L60-L62?

File: dags/openshift_nightlies/dag.py [hunk 3]
```diff
@@ -194,6 +196,43 @@ def build(self):
...
+
+        with TaskGroup("utils", prefix_group_id=False, dag=self.dag) as utils:
+            utils_tasks = self._get_scale_ci_diagnosis().get_utils()
+            chain(*utils_tasks)
+
+        with TaskGroup("benchmarks", prefix_group_id=False, dag=self.dag) as benchmarks:
+            benchmark_tasks = self._get_e2e_benchmarks().get_benchmarks()
+            chain(*benchmark_tasks)
+
+        initialize_cluster >> benchmarks >> utils
```
Comment (whitleykeith) [inline dags/openshift_nightlies/dag.py:231]: can you add the platform connector task to this dag now that's been merged?

File: dags/openshift_nightlies/dag.py [hunk 4]
```diff
@@ -194,6 +196,43 @@ def build(self):
     def _get_openshift_installer(self):
         return rogcp.RoGCPInstaller(self.dag, self.config, self.release)
 
+class PrebuiltOpenshiftNightlyDAG(AbstractOpenshiftNightlyDAG):
+    def __init__(self,openshift_release, DagConfig):
```
Comment (whitleykeith) [inline dags/openshift_nightlies/dag.py:200]: DagConfig is a class name but should be a param, i think you were meaning to type hint like `dag_config: DagConfig`

**Review by whitleykeith — CHANGES_REQUESTED** [review]

File: dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py (hunk 1 above)
Comment (Harshith-umesh) [reply dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py]: yes removed.

File: dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py (hunk 2 above)
Comment (Harshith-umesh) [reply dags/openshift_nightlies/tasks/install/prebuilt/initialize_cluster.py]: removed.

File: dags/openshift_nightlies/dag.py (hunk 3 above)
Comment (Harshith-umesh) [reply dags/openshift_nightlies/dag.py]: done!

File: dags/openshift_nightlies/dag.py (hunk 4 above)
Comment (Harshith-umesh) [reply dags/openshift_nightlies/dag.py]: right

File: dags/openshift_nightlies/manifest.yaml [hunk 5]
```diff
@@ -133,3 +133,17 @@ platforms:
         config:
           install: rogcp/sdn.json
           benchmarks: data-plane.json
+  prebuilt:
+    versions: ["4.10"]
```
Comment (whitleykeith) [inline dags/openshift_nightlies/manifest.yaml:137]: We can't control openshift versions for prebuilts so we should probably just make this 4.x or remove this field entirely if we can. This key is only used for release streams and naming the dag so we can use whatever we want

**Review by whitleykeith — CHANGES_REQUESTED** [review]

File: dags/openshift_nightlies/util/manifest.py [hunk 6]
```diff
@@ -172,6 +172,33 @@ def get_rogcp_releases(self):
...
+        prebuilt = self.yaml['platforms']['prebuilt']
+        for version in self.yaml['versions']:
+            if version['version'] in prebuilt['versions']:
+                version_number = version['version']
+                release_stream = version['releaseStream']
+                version_alias = version['alias']
+                for variant in prebuilt['variants']:
+                    release = OpenshiftRelease(
+                        platform="prebuilt",
+                        version=version_number,
```
[the remark this answers is not in the document]
Comment (Harshith-umesh) [reply dags/openshift_nightlies/util/manifest.py:185]: 4.X it is!

**Review by whitleykeith — APPROVED** [review]
LGTM

File: dags/openshift_nightlies/scripts/install/prebuilt.sh [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    echo ${user}; echo ${pass}; echo ${url};     
+    curl -sS https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/openshift-client-linux.tar.gz | tar xz oc
+    export PATH=$PATH:$(pwd)
+    
+    mkdir workspace
+    cd workspace 
+    export KUBECONFIG=kubeconfig
+    oc login -u ${user} -p ${pass} ${url} --insecure-skip-tls-verify
+    
+    unset KUBECONFIG
```
Comment (dry923) [inline dags/openshift_nightlies/scripts/install/prebuilt.sh:23]: why are we exporting the kubeconfig not using it then unsetting it 2 lines later? Or am I missing something?

**Review by morenod — APPROVED** [review]
LGTM

**Review by dry923 — APPROVED** [review]
LGTM!

**Pull request merged**